src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -216,8 +216,22 @@ void DatabaseCatalog::shutdownImpl()
 
     /// We still hold "databases" (instead of std::move) for Buffer tables to flush data correctly.
 
+    /// Delay shutdown of temporary and system databases. They will be shutdown last.
```

Comment:
Can we explain why?

Reply:
Added a comment https://github.com/ClickHouse/ClickHouse/pull/51517